stellar-contract-env-host/src/host.rs

```diff
@@ -1719,20 +1683,11 @@ impl CheckedEnv for Host {
     }
 
     fn binary_slice(&self, b: Object, start: RawVal, end: RawVal) -> Result<Object, HostError> {
-        let start = self.usize_from_rawval_u32_input("start", start)?;
-        let end = self.usize_from_rawval_u32_input("end", end)?;
+        let start = self.u32_from_rawval_input("start", start)?;
+        let end = self.u32_from_rawval_input("end", end)?;
         let vnew = self.visit_obj(b, move |hv: &Vec<u8>| {
-            if start > hv.len() {
```

Comment:
This was a bug. Would've caused a slice out of bound panic if passing in `hv.len()`. 
Should be `start >= hv.len()`